ssl/statem/statem_lib.c

```diff
@@ -646,12 +646,11 @@ MSG_PROCESS_RETURN tls_process_key_update(SSL *s, PACKET *pkt)
     return MSG_PROCESS_FINISHED_READING;
 }
 
-#ifndef OPENSSL_NO_NEXTPROTONEG
 /*
  * ssl3_take_mac calculates the Finished MAC for the handshakes messages seen
  * to far.
  */
-static void ssl3_take_mac(SSL *s)
+void ssl3_take_mac(SSL *s)
```

Comment:
Removed code was checking return value from final_finish_mac, this one doesn't. Is it checked elsewhere?

Reply:
Good point. I added the missing check.